code/src/WijmoProvider/Features/Pagination.ts

```diff
@@ -191,6 +194,30 @@ namespace WijmoProvider.Feature {
             }
         }
 
+        public getCurrentPage(): OSFramework.OSStructure.ReturnMessage {
+            let isSuccess = true;
+            let value = this.pageIndex;
+            let message = '';
+            let code: OSFramework.Enum.ErrorCodes;
+
+            // we don't want to return page index if there is server side pagination
+            if (this._grid.config.serverSidePagination) {
+                isSuccess = false;
+                value = 0;
+                message = this._serverSideOnErrorMessage;
+                code =
+                    OSFramework.Enum.ErrorCodes
+                        .API_FailedPaginationGetCurrentPage;
+            }
+
+            return {
+                code,
+                isSuccess,
+                value,
+                message
+            };
+        }
+
```

Comment:
I feel that this code is very strange... maybe it should be in the API level... not here..

Reply:
I had the same feeling as well. I guess I opted for the wrong choice. Will fix it.